libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/ListItem.kt

```diff
@@ -117,25 +119,40 @@ fun ListItem(
     val decoratedTrailingContent: (@Composable () -> Unit)? = trailingContent?.let { content ->
         {
             CompositionLocalProvider(
-                LocalContentColor provides ElementTheme.materialColors.primary,
+                LocalContentColor provides leadingTrailingContentColor,
+                LocalTextStyle provides ElementTheme.typography.fontBodyMdRegular,
             ) {
                 content()
             }
         }
     }
+
+    val setUpModifier = modifier
+        .then(if (isCompact) Modifier else Modifier.padding(vertical = 4.dp))
+        .clickable(enabled = onClick != null, onClick = onClick ?: {})
```

Comment:
Maybe also take into account the `enabled` parameter? I do not know if we want to have a disabled ListItem still clickable. Maybe...

```
.clickable(enabled = enabled && onClick != null, onClick = onClick ?: {})
```

Reply:
Yes, you're right. It depends on the implementation, but here `enabled` is up to us to apply, so it should be taken into account.